ext/pdo_mysql/tests/pdo_mysql___construct_uri.phpt

```diff
@@ -38,8 +38,9 @@ MySQLPDOTest::skip();
             }
 
             if ($fp = @fopen($file, 'w')) {
-                fwrite($fp, sprintf('mysql:dbname=letshopeinvalid;%s%s',
-                    chr(0), $dsn));
+                $dsnUnknownDatabase = preg_replace('~dbname=[^;]+~', 'dbname=letshopeinvalid', $dsn)
+                    . chr(0) . ';host=nonsense;unix_socket=nonsense';
+                fwrite($fp, $dsnUnknownDatabase);
```

Comment:
It seems to me that this change defeats the purpose of that part of the test. The test is for checking that a NUL byte terminates the DSN string. If there is no NUL byte (or it would be ignored), later options override previous options (`dbname=good;dbname=bad` would be the same as `dbname=bad`).

Reply:
I fixed the test in a minimalistic way.

I aggre, we can basically copy the 1st test with `fwrite($fp, $dsn. chr(0) . ';host=nonsense;unix_socket=nonsense');`, not bother with `dbname=letshopeinvalid` and expect the same result. Do you want me to do that?

Reply:
> I aggre, we can basically copy the 1st test with `fwrite($fp, $dsn. chr(0) . ';host=nonsense;unix_socket=nonsense');`, not bother with `dbname=letshopeinvalid` and expect the same result.

That make sense to me.

Reply:
done, 1st test vs chr(0) test:

![image](https://user-images.githubusercontent.com/2228672/166211444-303f85a5-ba27-45e9-bdae-68083abded90.png)

Reply:
CI issue unrelated, let's merge this PR, needed for https://github.com/php/php-src/pull/8392